mini-sntp/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#![deny(rust_2018_idioms)]
+#![warn(clippy::all, clippy::pedantic)]
+#![allow(
+    clippy::doc_markdown,
+    clippy::missing_errors_doc,
+    clippy::module_name_repetitions,
+    clippy::must_use_candidate,
+    clippy::too_many_lines,
+    clippy::use_self,
+    clippy::unusual_byte_groupings
+)]
+
+use std::net::{SocketAddr, ToSocketAddrs, UdpSocket};
+
+mod error;
+pub use error::{BadServerResponseReason, Error};
+
+/// The result of [`query`]
+#[derive(Debug)]
+pub struct SntpTimeQueryResult {
+    pub local_clock_offset: chrono::Duration,
+    pub round_trip_delay: chrono::Duration,
+}
+
+/// Executes an SNTP query against the NTPv3 server at the given address.
+///
+/// Ref: <https://tools.ietf.org/html/rfc2030>
+pub fn query<A>(addr: &A) -> Result<SntpTimeQueryResult, Error>
+where
+    A: ToSocketAddrs,
+{
+    let addr = addr
+        .to_socket_addrs()
+        .map_err(|err| Error::ResolveNtpPoolHostname(Some(err)))?
+        .next()
+        .ok_or(Error::ResolveNtpPoolHostname(None))?;
+
+    let socket = UdpSocket::bind("0.0.0.0:0").map_err(Error::BindLocalSocket)?;
+    socket
+        .set_read_timeout(Some(std::time::Duration::from_secs(10)))
+        .map_err(Error::SetReadTimeoutOnSocket)?;
+    socket
+        .set_write_timeout(Some(std::time::Duration::from_secs(10)))
+        .map_err(Error::SetWriteTimeoutOnSocket)?;
+
+    let mut num_retries_remaining = 3;
+    loop {
+        match query_inner(&socket, addr) {
+            Ok(result) => return Ok(result),
+            Err(err) => {
+                let is_retriable = match &err {
+                    Error::SendClientRequest(err) | Error::ReceiveServerResponse(err) => {
+                        err.kind() == std::io::ErrorKind::TimedOut || // Windows
```

Comment:
Don't need this check any more.

Reply:
https://github.com/Azure/iot-identity-service/pull/83#discussion_r539954457

Reply:
I prefer to remove it. It's out-of-place in a repository that otherwise has no Windows support.

Reply:
Are you sure? It's just one line, and it's not really a massive "maintenance burden" leaving it in. Plus, while the fix is ostensibly windows specific, there is no reason some other platform couldn't return the same error at some point (highly unlikely as it is).

Reply:
If you want to keep it that's fine, but remove the `// Windows` and `// Unix` comments.

Reply:
Cool, will do :+1: